lib/utils.js

```diff
@@ -37,13 +37,31 @@ export function through (transform) {
   })
 }
 
-export function bulk (fn) {
-  return new Writable({
-    objectMode: true,
-    writev: function (chunks, cb) {
-      fn(chunks.map(chunk => chunk.chunk), cb)
+export function retimer (fn, ms) {
+  const timeout = setTimeout(fn, ms)
+  return {
+    reschedule (newMs) {
+      timeout.refresh(newMs)
+    },
+    clear () {
+      clearTimeout(timeout)
     }
-  })
+  }
+}
+
+export async function * batch (readable, fn, batchSize) {
+  let chunks = []
+  for await (const chunk of readable) {
+    chunks.push(fn(chunk))
+    if (chunks.length === batchSize) {
+      yield (chunks)
+      chunks = []
+    }
+  }
+  // if chunks is half full when the iterator ends
+  if (chunks.length > 0) {
+    yield chunks
+  }
```

Comment:
Can you benchmark the simple implementation against https://github.com/mcollina/hwp/blob/main/index.js? I have a feeling this might problematic (or maybe V8 just got way faster over the years).